src/db.rs

```diff
@@ -1753,46 +1788,39 @@ impl DBWithThreadMode<MultiThreaded> {
     /// Creates column family with given name and options
     pub fn create_cf<N: AsRef<str>>(&self, name: N, opts: &Options) -> Result<(), Error> {
         let inner = self.create_inner_cf_handle(name.as_ref(), opts)?;
-        self.cfs
-            .cfs
-            .write()
-            .unwrap()
-            .insert(name.as_ref().to_string(), ColumnFamily { inner });
+        self.cfs.cfs.write().unwrap().insert(
+            name.as_ref().to_string(),
+            Arc::new(UnboundColumnFamily { inner }),
+        );
         Ok(())
     }
 
     /// Drops the column family with the given name by internally locking the inner column
     /// family map. This avoids needing `&mut self` reference
     pub fn drop_cf(&self, name: &str) -> Result<(), Error> {
-        let inner = self.inner;
         if let Some(cf) = self.cfs.cfs.write().unwrap().remove(name) {
```

Comment:
I'd replace here `unwrap()` for something like:
```rust
.map_err(|e| Error::new(e.to_string()))?
```

Reply:
see my comment: https://github.com/rust-rocksdb/rust-rocksdb/pull/509/files#r674138005